apm-agent-core/src/main/java/co/elastic/apm/agent/impl/circuitbreaker/CircuitBreaker.java

```diff
@@ -0,0 +1,144 @@
+        pollInterval = circuitBreakerConfiguration.getStressMonitoringPollingInterval();
+        threadPool = ExecutorUtils.createSingleThreadDeamonPool("circuit-breaker", 1);
+    }
+
+    @Override
+    public void start(ElasticApmTracer tracer) {
+        // failsafe loading of stress monitors in isolation
+        loadGCStressMonitor(tracer);
+
+        threadPool.execute(new Runnable() {
+            @Override
+            public void run() {
+                pollStressMonitors();
+            }
+        });
+    }
+
+    private void loadGCStressMonitor(ElasticApmTracer tracer) {
+        try {
+            stressMonitors.add(new GCStressMonitor(tracer));
+        } catch (Throwable throwable) {
+            logger.error("Failed to load the GC stress monitor. Circuit breaker will not be triggered based on GC events.", throwable);
+        }
+    }
+
+    private void pollStressMonitors() {
+        while (!Thread.currentThread().isInterrupted()) {
+            try {
+                if (circuitBreakerConfiguration.isCircuitBreakerEnabled()) {
+                    if (isCurrentlyUnderStress) {
+                        boolean stressRelieved = true;
+                        for (StressMonitor stressMonitor : stressMonitors) {
+                            try {
+                                stressRelieved &= stressMonitor.isStressRelieved();
+                            } catch (Exception e) {
+                                if (logger.isDebugEnabled()) {
+                                    logger.debug("Failed to poll " + stressMonitor.getClass().getName(), e);
+                                }
+                            }
+                        }
+                        if (stressRelieved) {
+                            isCurrentlyUnderStress = false;
+                            tracer.stressRelieved();
+                        }
+                    } else {
+                        for (StressMonitor stressMonitor : stressMonitors) {
+                            try {
+                                if (stressMonitor.isUnderStress()) {
+                                    isCurrentlyUnderStress = true;
+                                    tracer.stressDetected();
+                                    break;
+                                }
+                            } catch (Exception e) {
+                                if (logger.isDebugEnabled()) {
+                                    logger.debug("Failed to poll " + stressMonitor.getClass().getName(), e);
+                                }
+                            }
+                        }
+                    }
+                } else if (isCurrentlyUnderStress) {
+                    isCurrentlyUnderStress = false;
+                    tracer.stressRelieved();
+                }
+            } catch (Throwable throwable) {
+                logger.error("Error occurred during Circuit Breaker polling", throwable);
```

Comment:
this catch block should never be called, because all potential exceptions from stress monitor are already caught.

Reply:
All potential `Exception`s but not `Throwable`s. 
If, by accident, we get an `OutOfMemoryError` in our monitor, this is probably the only agent functionality I want to keep alive 🙂 .
This is the thread's main loop, it makes sense to not let the thread die from unnatural causes.

Reply:
Ok then probably makes sense to at least add a comment why we need to catch everything here.